### PR TITLE
Imdb

### PR DESCRIPTION
**Pull request by tomkoren21**
Add the `tensorleap` and local code for IMDB

### CODE REVIEW

File: examples/imdb/imdb/.tensorleap/dataset.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    cloud_path = join("assets", "tokenizer.json")
+    local_path = _download(cloud_path)
+    tokenizer = load_tokanizer(local_path)
+    print("finished downloading assets")
+    return tokenizer, train_dict
+
+
+def create_input(train_dict: dict, half_t_size: int, half_v_size: int) -> (List[str], List[List[float]],
+                                                                           List[List[float]], List[str],
+                                                                           List[List[float]], List[List[float]]):
```
Comment (yotamazriel) [inline examples/imdb/imdb/.tensorleap/dataset.py:83]: What about using `dataclass` here?

File: examples/imdb/imdb/requirement.yaml [hunk 2]
```diff
@@ -0,0 +1,115 @@
+absl-py==0.12.0
+astunparse==1.6.3
+attrs==21.4.0
+cached-property==1.5.2
```
Comment (doronHarnoy) [inline examples/imdb/imdb/requirement.yaml]: Are you sure its a yaml file? looks more like a txt file to me.
Another one, do we need to publish it if we publish the poetry pyproject.toml?

File: examples/imdb/imdb/imdb/model_infer.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        layers.Dense(28, activation='relu'),
+        layers.GlobalAveragePooling1D(),
+        layers.Dropout(0.2),
+        layers.Dense(2, activation='sigmoid')])
+    x = vectorized_inputs
+    for lr in model.layers:
+        x = lr(x)
+    output = layers.Softmax(axis=-1)(x)
+    tl_model = tf.keras.Model(inputs=vectorized_inputs, outputs=output)
+    return tl_model
```
Comment (doronHarnoy) [inline examples/imdb/imdb/imdb/model_infer.py:33]: lets move the creation of the model here to functional and fix the last layer to be only softmax

File: examples/imdb/imdb/.tensorleap/dataset.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    local_path = _download(comment_path)
+    with open(local_path, 'r') as f:
+        comment = f.read()
+    tokenizer = subset.data['tokenizer']
+    padded_input = prepare_input(tokenizer, comment)
+    return padded_input
+
+
+def gt_sentiment(idx: int, subset: Union[SubsetResponse, list]) -> List[float]:
+    print("gt_sentiment")
```
Comment (doronHarnoy) [inline examples/imdb/imdb/.tensorleap/dataset.py:154]: Do you think that we need the prints in the dataset function here?

File: examples/imdb/imdb/imdb/model_infer.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras import layers
+from tensorflow.keras.preprocessing.text import tokenizer_from_json
+import json
+from utils import standartize
+from tensorflow.keras.preprocessing.sequence import pad_sequences
+max_features = 10000
+sequence_length = 250
+AUTOTUNE = tf.data.AUTOTUNE
```
Comment (doronHarnoy) [inline examples/imdb/imdb/imdb/model_infer.py:15]: who is using this?

**Review by doronHarnoy — APPROVED** [review]
Looking Good!

File: examples/imdb/imdb/requirement.yaml (hunk 2 above)
Comment (tomkoren21) [reply examples/imdb/imdb/requirement.yaml]: `requirement.yaml` is removed in the final commit

File: examples/imdb/imdb/imdb/model_infer.py (hunk 5 above)
Comment (tomkoren21) [reply examples/imdb/imdb/imdb/model_infer.py]: line 15 is removed, rest is used

File: examples/imdb/imdb/.tensorleap/dataset.py (hunk 4 above)
Comment (tomkoren21) [reply examples/imdb/imdb/.tensorleap/dataset.py]: all prints removed

File: examples/imdb/imdb/imdb/model_infer.py (hunk 3 above)
Comment (tomkoren21) [reply examples/imdb/imdb/imdb/model_infer.py]: This is done in [f98ef1f](https://github.com/tensorleap/tensorleap/pull/8/commits/f98ef1f091b6d63e3d85be0b974f9e9c7f176d9d)

**Review by yotamazriel — APPROVED** [review]
LGTM
